docs(bpmn): add AI Agent Sub-process reference to ad-hoc sub-processes page (#9237)
camunda/camunda-docs: docs/components/modeler/bpmn/ad-hoc-subprocesses/ad-hoc-subprocesses.md

---
id: ad-hoc-subprocesses
title: "Ad-hoc sub-processes"

> description: "With ad-hoc sub-processes, you can model a flexible set of BPMN activities that activate in any order, run multiple times, or be skipped."

Reviewer: This description reads awkwardly to me. I think there are three reasons for this:

- "activities that activate" feels like it needs a different verb.
- "be skipped" is passive, but "activities that activate" and "activities that run multiple times" are active.
- The list is ordered longest element to shortest. "activate in any order" > "run multiple times" > "be skipped"

I would consider a rewrite, but I'm not exactly sure of the technical details, so take this suggestion as merely a starting point:

```suggestion
description: "With ad-hoc sub-processes, you can model a flexible set of BPMN activities that execute in any order, any number of times."
```

Author: Addressed!

---

With ad-hoc sub-processes, you can model a flexible set of BPMN activities that activate in any order, run multiple times, or be skipped.

:::info
With the [AI Agent Sub-process connector](/components/connectors/out-of-the-box-connectors/agentic-ai-aiagent-subprocess.md), you can implement an AI agent that dynamically selects and invokes tools, using the [job worker implementation](#job-worker-implementation) of an ad-hoc sub-process.
:::

## About

> Ad-hoc sub-processes are a special kind of [embedded subprocesses](../embedded-subprocesses/embedded-subprocesses.md) with an **ad-hoc marker** (represented

Reviewer: Should we take this moment to also update this?

```suggestion
Ad-hoc sub-processes are a special kind of [embedded subprocess](../embedded-subprocesses/embedded-subprocesses.md) with an **ad-hoc marker** (represented
```
